### PR TITLE
Feat/ai fix feedback command [IDE-634]

### PR DESCRIPTION
**Pull request by acke**
### Description

Track the behavior of the users regarding the Fix AI.

The events to track:
- users click button “Apply fix“
- users click thumbs up or down

### Checklist

- [x] Tests added and all succeed
- [x] Linted

### CODE REVIEW

File: infrastructure/code/issue_enhancer.go [hunk 1]
```diff
@@ -251,16 +251,16 @@ func ToEncodedNormalizedPath(rootPath string, filePath string) (string, error) {
...
 			Title:     types.CodeSubmitFixFeedback,
 			CommandId: types.CodeSubmitFixFeedback,
-			Arguments: []any{fixId, positive},
+			Arguments: []any{fixId, feedback},
 		}
 	}
 	actionCommandMap := data_structure.NewOrderedMap[types.MessageAction, types.CommandData]()
-	positiveFeedbackCmd := createCommandData(true)
-	negativeFeedbackCmd := createCommandData(false)
+	positiveFeedbackCmd := createCommandData("FIX_POSITIVE_FEEDBACK")
```
Comment (bastiandoetsch) [inline infrastructure/code/issue_enhancer.go:262]: Let's have these values as constants.

File: domain/ide/command/code_fix_feedback.go [hunk 2]
```diff
@@ -38,8 +38,8 @@ func (cmd *codeFixFeedback) Command() types.CommandData {
 func (cmd *codeFixFeedback) Execute(ctx context.Context) (any, error) {
 	args := cmd.command.Arguments
 	fixId := args[0].(string)
-	positive := args[1].(bool)
-	err := cmd.apiClient.SubmitAutofixFeedback(ctx, fixId, positive)
+	feedback := args[1].(string)
+	err := cmd.apiClient.SubmitAutofixFeedback(ctx, fixId, feedback)
```
Comment (bastiandoetsch) [inline domain/ide/command/code_fix_feedback.go:42]: Let's change this to invoke a go routine.

Comment (bastiandoetsch) [reply domain/ide/command/code_fix_feedback.go:42]: Also, let's also analyse, if this needs to be added and respected when using globalIgnoresFlow and code-client-go

File: infrastructure/code/issue_enhancer.go (hunk 1 above)
Comment (acke) [reply infrastructure/code/issue_enhancer.go]: Fixed!

File: domain/ide/command/code_fix_feedback.go (hunk 2 above)
Comment (acke) [reply domain/ide/command/code_fix_feedback.go]: Go routine fixed.

Comment (acke) [reply domain/ide/command/code_fix_feedback.go]: So, I have been looking through code-client-go and Go Application Framework, trying to find anything indicating that they support autofix/ai fix, and that I would need to add support for ai fix feedback. But I could not find anything on autofix.

I also did code scans with ignores enabled, towards dev16, and applied AI fixes. I successfully saved feedback for the issues.

**Review by bastiandoetsch — APPROVED** [review]

File: domain/ide/command/code_fix_feedback_test.go [hunk 3]
```diff
@@ -44,28 +53,15 @@ func Test_codeFixFeedback_SubmittedSuccessfully(t *testing.T) {
...
-		},
-		apiClient: &apiClient,
-	}
 
-	_, err := codeFixFeedbackCmd.Execute(context.Background())
-	assert.Error(t, err)
-	assert.False(t, apiClient.feedbackSubmitted)
+	assert.NoError(t, err)
+	assert.Eventually(t, func() bool {
+		return apiClient.feedbackSubmitted != ""
```
[the remark this answers is not in the document]
Comment (acke) [reply domain/ide/command/code_fix_feedback_test.go:65]: Fixed!

**Pull request merged**